h2o-core/src/main/java/water/DKVManager.java

```diff
@@ -111,10 +112,17 @@ protected void setupLocal() {
         }
         if(value.isNull()) continue;
 
-        if (value.isFrame()) {
-          ((Frame) value.get()).retain(removalFutures, retainedKeys);
-        } else if (value.isModel()) {
-          ((Model)value.get()).remove(removalFutures);
+        try {
+          TypeMap.getTheFreezableOrThrow(value.type());
+          if (value.isFrame()) {
+            ((Frame) value.get()).retain(removalFutures, retainedKeys);
+          } else if (value.isModel()) {
+            ((Model) value.get()).remove(removalFutures);
+          }
+
+        } catch (ClassNotFoundException e) {
```

Comment:
Please add a comment on why is this exception caught.

Reply:
Done-.